spec/infinispan_local_spec.js

```diff
@@ -5,370 +5,274 @@ var readFile = Promise.denodeify(require('fs').readFile);
 var f = require('../lib/functional');
 var t = require('./utils/testing'); // Testing dependency
 
-describe('Infinispan local client', function() {
-  var client = t.client(t.local);
+function describeInfinispanBehaviourInLocalModeWithDifferentHotrodProtocol(options) {
+  describe('with hotrod protocol: ' + options.version, function () {
+    var client = t.client(t.local, null, options.version);
 
-  beforeEach(function(done) { client
```

Comment:
Implemented this as we have agreed. Also placed a check for iterator tests so that if the protocol is not 2.5, then they are not run.